[Flaky Test] -  TestDownloadBodyError

## Flaky Test

* **Test Name:** TestDownloadBodyError
* **Link:** [Link to file/line number in github.](https://github.com/elastic/elastic-agent/blob/main/internal/pkg/artifact/download/http/downloader_test.go#L28)
* **Branch:** main
* **Notes:** It happens quite often, can be reproduced locally (Windows 11) 9/10 times

### Stack Trace

```
=== RUN   TestDownloadBodyError
    downloader_test.go:69: 
        	Error Trace:	downloader_test.go:69
        	Error:      	"[{download progress from %s has fetched %s @ %sps [http://127.0.0.1:65283/beats/filebeat/filebeat-7.5.1-linux-x86_64.tar.gz 0B NaNB]} {download from %s failed at %s @ %sps: %s [http://127.0.0.1:65283/beats/filebeat/filebeat-7.5.1-linux-x86_64.tar.gz 0B NaNB %!s(*errors.errorString=&{unexpected EOF})]}]" should have 1 item(s), but has 2
        	Test:       	TestDownloadBodyError
        	Messages:   	download error not logged at info level
--- FAIL: TestDownloadBodyError (0.00s)

FAIL

Process finished with the exit code 1

```

Line https://github.com/elastic/elastic-agent/blob/6df928dd1c1d8e36008bf54f624d8174f7a86c19/internal/pkg/artifact/download/http/downloader.go#L165 does not return an error or failing status code as it should.

--- a/internal/pkg/artifact/download/http/downloader_test.go
+++ b/internal/pkg/artifact/download/http/downloader_test.go
@@ -64,10 +64,10 @@ func TestDownloadBodyError(t *testing.T) {
 		t.Fatal("expected Download to return an error")
 	}
 
-	require.Len(t, log.info, 1, "download error not logged at info level")
-	assert.Equal(t, log.info[0].record, "download from %s failed at %s @ %sps: %s")
-	require.Len(t, log.warn, 1, "download error not logged at warn level")
-	assert.Equal(t, log.warn[0].record, "download from %s failed at %s @ %sps: %s")
+	require.GreaterOrEqual(t, len(log.info), 1, "download error not logged at info level")
+	assert.Equal(t, log.info[len(log.info)-1].record, "download from %s failed at %s @ %sps: %s")
+	require.GreaterOrEqual(t, len(log.warn), 1, "download error not logged at warn level")
+	assert.Equal(t, log.warn[len(log.warn)-1].record, "download from %s failed at %s @ %sps: %s")
 }
 
 func TestDownloadLogProgressWithLength(t *testing.T) {

